common/lazy-pages/src/lib.rs

```diff
@@ -44,29 +48,47 @@ fn mprotect_lazy_pages(mem: &mut impl Memory, protect: bool) {
 }
 
 /// Try to enable and initialize lazy pages env
-pub fn try_to_enable_lazy_pages(pages_final_prefix: [u8; 32]) -> bool {
-    gear_ri::init_lazy_pages(pages_final_prefix)
+pub fn try_to_enable_lazy_pages(prefix: [u8; 32]) -> bool {
+    let ctx = LazyPagesRuntimeContext {
+        page_sizes: vec![WasmPage::size(), GearPage::size()],
+        global_names: vec![
+            GLOBAL_NAME_GAS.to_string(),
+            GLOBAL_NAME_ALLOWANCE.to_string(),
+        ],
+        pages_storage_prefix: prefix.to_vec(),
+    };
+
+    gear_ri::init_lazy_pages(ctx)
 }
 
 /// Protect and save storage keys for pages which has no data
 pub fn init_for_program(
     mem: &mut impl Memory,
     program_id: ProgramId,
     stack_end: Option<WasmPage>,
-    globals_config: GlobalsConfig,
-    lazy_pages_weights: LazyPagesWeights,
+    globals_config: GlobalsAccessConfig,
+    weights: LazyPagesWeights,
 ) {
-    let wasm_mem_addr = mem.get_buffer_host_addr();
-    let wasm_mem_size = mem.size();
-    let program_id = <[u8; 32]>::from(program_id.into_origin()).into();
+    let weights = [
+        weights.signal_read,
+        weights.signal_write,
+        weights.signal_write_after_read,
+        weights.host_func_read,
+        weights.host_func_write,
+        weights.host_func_write_after_read,
+        weights.load_page_storage_data,
+    ]
+    .iter()
+    .map(|w| w.one())
+    .collect();
```

Comment:
No need to create an iterator for arrays.
```suggestion
    .map(|w| w.one())
    .to_vec();
```

Reply:
done